### PR TITLE
Issue #36 fix null comparisons

### PR DESCRIPTION
**Pull request by costea32**
adding NullComparison helper class for consistency in all expected/actual null checks. Updated current strategies to use the null comparison, added NullComparisonResult that contains an extra property to indicate that comparison should stop(null/null case).

### CODE REVIEW

File: src/main/java/org/testmonkeys/jentitytest/comparison/util/NullComparison.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package org.testmonkeys.jentitytest.comparison.util;
+
+import org.testmonkeys.jentitytest.comparison.ComparisonContext;
+
+public class NullComparison {
```
Comment (monofrei) [inline src/main/java/org/testmonkeys/jentitytest/comparison/util/NullComparison.java]: rename to NullComparator

File: src/main/java/org/testmonkeys/jentitytest/comparison/util/NullComparisonResult.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+public class NullComparisonResult extends ComparisonResult {
+
+    private boolean stopComparison;
+
+    public NullComparisonResult(boolean passed, ComparisonContext context, Object actual, Object expected) {
+        super(passed, context, actual, expected);
+    }
+
+    public boolean isStopComparison() {
```
Comment (monofrei) [inline src/main/java/org/testmonkeys/jentitytest/comparison/util/NullComparisonResult.java]: rename to stopComparison

File: src/main/java/org/testmonkeys/jentitytest/comparison/util/NullComparison.java (hunk 1 above)
Comment (monofrei) [inline src/main/java/org/testmonkeys/jentitytest/comparison/util/NullComparison.java]: rename to NullComparator

Comment (costea32) [inline src/main/java/org/testmonkeys/jentitytest/comparison/util/NullComparison.java]: done

File: src/main/java/org/testmonkeys/jentitytest/comparison/util/NullComparisonResult.java (hunk 2 above)
Comment (costea32) [inline src/main/java/org/testmonkeys/jentitytest/comparison/util/NullComparisonResult.java]: done

**Comment by costea32** [comment]
Fixed pull request comments

**Pull request merged**